psdk_wrapper/src/modules/perception.cpp

```diff
@@ -0,0 +1,404 @@
+  return_code = DjiPerception_GetStereoCameraParameters(&cameraParametersPacket);
+  if (return_code != DJI_ERROR_SYSTEM_MODULE_CODE_SUCCESS)
+  {
+    RCLCPP_ERROR(get_logger(),
+                "Get camera parameters failed, Error code: %ld",
+                return_code);
+  }
+  perception_camera_parameters_msg.header.stamp = this->get_clock()->now();
+  perception_camera_parameters_msg.header.frame_id = "stereo_cameras_parameters_link";
+  perception_camera_parameters_msg.stereo_cameras_direction = stereo_cameras_direction_;
+
+  // Populate the message fields
+  std::copy(std::begin(cameraParametersPacket
+                           .cameraParameters[stereo_cameras_direction_]
+                           .leftIntrinsics),
+            std::end(cameraParametersPacket
+                         .cameraParameters[stereo_cameras_direction_]
+                         .leftIntrinsics),
+            std::begin(perception_camera_parameters_msg.left_intrinsics));
+  std::copy(std::begin(cameraParametersPacket
+                           .cameraParameters[stereo_cameras_direction_]
+                           .rightIntrinsics),
+            std::end(cameraParametersPacket
+                         .cameraParameters[stereo_cameras_direction_]
+                         .rightIntrinsics),
+            std::begin(perception_camera_parameters_msg.right_intrinsics));
+  std::copy(
+      std::begin(
+          cameraParametersPacket.cameraParameters[stereo_cameras_direction_]
+              .rotationLeftInRight),
+      std::end(
+          cameraParametersPacket.cameraParameters[stereo_cameras_direction_]
+              .rotationLeftInRight),
+      std::begin(perception_camera_parameters_msg.rotation_left_in_right));
+  std::copy(
+      std::begin(
+          cameraParametersPacket.cameraParameters[stereo_cameras_direction_]
+              .translationLeftInRight),
+      std::end(
+          cameraParametersPacket.cameraParameters[stereo_cameras_direction_]
+              .translationLeftInRight),
+      std::begin(perception_camera_parameters_msg.translation_left_in_right));
+
+  perception_camera_parameters_pub_->publish(perception_camera_parameters_msg);
+
+  /*
+  * Please note that data type for left side stereo cameras are 1, 3, 5, 21, 23, 25.
+  * data type for right side stereo cameras are 2, 4, 6, 22, 24, 26.
+  * refer typedef enum E_DjiPerceptionCameraPosition.
+  * below logic is applied to keep only two topic one for left camera and another for right camera.
+  */ 
+  if (imageInfo.dataType % 2)
+  {
+    auto img = std::make_unique<sensor_msgs::msg::Image>();
+    img->height = imageInfo.rawInfo.height;
+    img->width = imageInfo.rawInfo.width;
+    img->step =  imageInfo.rawInfo.width;
+    img->encoding = "mono8";
+    img->data = std::vector<uint8_t>(imageRawBuffer, imageRawBuffer + bufferLen);
+    img->header.stamp = this->get_clock()->now();
+    img->header.frame_id = "stereo_cameras_link";
```

Comment:
you could consider making the frame id a parameter of the perception module so that users could be able to configure it's value, instead of putting it hardcoded